src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+<Sandpack>
+
+```js App.js
+import { useState, Suspense } from "react";
+import { Message } from "./message.js";
+
+function fetchMessage() {
+  return new Promise((resolve, reject) => {
+    setTimeout(reject, 3000, new Error("no new message found."));
+  }).catch((error) => {
+    console.log(error.toString());
+    return "no new message found.";
+  });
+}
+
+export default function App() {
+  const [messagePromise, setMessagePromise] = useState(null);
+  const [show, setShow] = useState(false);
+  function download() {
+    setMessagePromise(fetchMessage());
+    setShow(true);
+  }
+
+  if (show) {
+    return (
+      <Suspense fallback={<p>⌛Downloading message...</p>}>
+        <Message messagePromise={messagePromise} />
+      </Suspense>
+    );
+  } else {
+    return <button onClick={download}>Download new message</button>;
+  }
+}
+```
+
+```js message.js
+"use client";
+
+import { use } from "react";
+
+export function Message({ messagePromise }) {
+  const content = use(messagePromise);
+  return <p>Here is the message: {content}</p>;
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+---
+
+## Troubleshooting {/*troubleshooting*/}
+
+### I'm getting this error: "An unsupported type was passed to use" {/*im-getting-this-error-an-unsupported-type-was-passed-to-use*/}
+
+The `use` hook only supports [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) and React's [Context](/learn/passing-data-deeply-with-context) as parameters.
+
+### I'm getting this error: "Suspense Exception: This is not a real error!" {/*suspense-exception-error*/}
+
+You are either calling `use` outside of a React component or Hook function, or calling `use` in a try-catch block. If you are calling `use` inside a `try-catch` block wrap your component in an error boundary, or call the promise's `catch` to catch the error and resolve the promise with another value. [See these examples](#dealing-with-rejected-promises).
```

Comment:
nit: try–catch should use an en dash (and not be in backticks unless you split `try` and `catch`)

Reply:
Fixed!